utils/moduleVisitor.js

```diff
@@ -0,0 +1,109 @@
+"use strict"
+exports.__esModule = true
+
+/**
+ * Returns an object of node visitors that will call
+ * 'visitor' with every discovered module path.
+ *
+ * todo: correct function prototype for visitor
+ * @param  {Function(String)} visitor [description]
+ * @param  {[type]} options [description]
+ * @return {object}
+ */
+exports.default = function visitModules(visitor, options) {
+  // if esmodule is not explicitly disabled, it is assumed to be enabled
+  options = Object.assign({ esmodule: true }, options)
+
+  let ignoreRegExps = []
+  if (options.ignore != null) {
+    ignoreRegExps = options.ignore.map(p => new RegExp(p))
+  }
+
+  function checkSourceValue(source) {
+    if (source == null) return //?
+
+    // handle ignore
+    if (ignoreRegExps.some(re => re.test(source.value))) return
+
+    // fire visitor
+    visitor(source)
+  }
+
+  // for import-y declarations
+  function checkSource(node) {
+    checkSourceValue(node.source)
+  }
+
+  // for CommonJS `require` calls
+  // adapted from @mctep: http://git.io/v4rAu
+  function checkCommon(call) {
+    if (call.callee.type !== 'Identifier') return
+    if (call.callee.name !== 'require') return
+    if (call.arguments.length !== 1) return
+
+    const modulePath = call.arguments[0]
+    if (modulePath.type !== 'Literal') return
+    if (typeof modulePath.value !== 'string') return
+
+    checkSourceValue(modulePath)
```

Comment:
This function would also apply for `require` function calls when `require` is shadowed. So I think it would make sense to also check the scope where the reference come from.